internal/app/referenceclient/impl.go

```diff
@@ -136,6 +145,52 @@ func (i *invoker) unary(
 	}, nil
 }
 
+func (i *invoker) idempotentUnary(
```

Comment:
If the only difference is the request and response type, you could make a generic function that all three invoke.

Something with a signature like so (note that it must be a function, not a method, because methods in Go cannot be generic):
```go
// These first two types are generic constraints.
type pointerMsg[T any] interface {
	*T
	proto.Message
}

type payloadContainer[T any] interface {
	pointerMsg[T]
	GetPayload() *v1.ConformancePayload
}

func invokeUnary[ReqMsg pointerMsg[ReqT], RespMsg payloadContainer[RespT], ReqT, RespT any](
	ctx context.Context,
	req *v1.ClientCompatRequest,
	method func(context.Context, *connect.Request[ReqT]) (*connect.Response[RespT], error),
) (*v1.ClientResponseResult, error) {
	// use generic type to construct request
	ur := ReqMsg(new(ReqT))
	// ...

	// and also to invoke RPC
	connResp, err := method(ctx, connect.NewRequest(ur))
	if err != nil {
		return nil, err
	}
	resp := RespMsg(connResp.Msg)
	// ...
} 

// You can call the above like so to pass the particular RPC endpoint
// as a generic function parameter.
resp, err := invokeUnary(ctx, req, i.client.IdempotentUnary)
```

Reply:
You could always add a TODO to consolidate in the future. For now, I think it's okay that `idempotent` is different, so we really only replicate all of the logic 2x.

Reply:
Yep. Done.